In [1]:
import os, json
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
all_files = os.listdir('results')
cfgs = sorted([n for n in all_files if 'cfg' in n])
perfs = sorted([n[:-11]+'_performance_summary.csv' for n in cfgs])

In [3]:
list(zip(cfgs, perfs))

[('2020-01-10-07:5719_config.cfg',
  '2020-01-10-07:5719_performance_summary.csv'),
 ('2020-01-10-07:5721_config.cfg',
  '2020-01-10-07:5721_performance_summary.csv'),
 ('2020-01-10-07:5722_config.cfg',
  '2020-01-10-07:5722_performance_summary.csv'),
 ('2020-01-10-08:5109_config.cfg',
  '2020-01-10-08:5109_performance_summary.csv'),
 ('2020-01-10-08:5113_config.cfg',
  '2020-01-10-08:5113_performance_summary.csv'),
 ('2020-01-10-08:5613_config.cfg',
  '2020-01-10-08:5613_performance_summary.csv'),
 ('2020-01-10-10:3539_config.cfg',
  '2020-01-10-10:3539_performance_summary.csv'),
 ('2020-01-10-10:3545_config.cfg',
  '2020-01-10-10:3545_performance_summary.csv'),
 ('2020-01-10-10:4801_config.cfg',
  '2020-01-10-10:4801_performance_summary.csv'),
 ('2020-01-10-12:5832_config.cfg',
  '2020-01-10-12:5832_performance_summary.csv'),
 ('2020-01-10-12:5842_config.cfg',
  '2020-01-10-12:5842_performance_summary.csv'),
 ('2020-01-10-12:5846_config.cfg',
  '2020-01-10-12:5846_performance_summary

In [4]:
pd.read_csv('results/'+perfs[0])

dataset     auc     acc  dice/F1  dice/F1_from_bin    spec    sens
0  DRIVE_train  0.9793  0.9551   0.8224            0.8209  0.9741  0.8236
1    DRIVE_val  0.9825  0.9545   0.8175            0.8045  0.9711  0.8354
2        DRIVE  0.9787  0.9555   0.8182            0.8189  0.9802  0.7861
3        STARE  0.9638  0.9548   0.7789            0.7686  0.9760  0.7711
4      CHASEDB  0.9635  0.9465   0.7248            0.7344  0.9746  0.6966
5      AV-WIDE  0.7965  0.9491   0.4276            0.5118  0.9965  0.2851

In [5]:
all_results = []
for i in range(len(cfgs)):
    with open('results/'+cfgs[i], 'r') as f:
        args = json.load(f)
    res = pd.read_csv('results/'+perfs[i])    
    all_results.append([args,res])

In [6]:
list(all_results[0][0].keys())

['layers',
 'n_classes',
 'in_channels',
 'up_mode',
 'pool_mode',
 'conv_bridge',
 'shortcut',
 'lr',
 'disable_transforms',
 'eps',
 'batch_size',
 'data_aug',
 'normalize',
 'metric',
 'patience',
 'scheduler_f',
 'n_epochs',
 'data_path',
 'end2end',
 'experiment_path',
 'config_file_path']

In [7]:
both_of_them_l = []
conv_bridge_l = []
shortcut_l = []
none_of_them_l = []
for i in range(len(cfgs)):
    conv_bridge = all_results[i][0]['conv_bridge']
    shortcut = all_results[i][0]['shortcut']
    lr = all_results[i][0]['lr']
    val_auc = all_results[i][1].iat[1,1]
    train_auc = all_results[i][1].iat[0,1]
    test_auc = all_results[i][1].iat[2,1]

    
    if conv_bridge and shortcut: 
        print('this one for both of them!', lr)
        print(all_results[i][1])
    if conv_bridge and shortcut:
        both_of_them_l.append([lr,val_auc,test_auc])
    if conv_bridge and not shortcut:
        conv_bridge_l.append([lr,val_auc,test_auc])
    if not conv_bridge and shortcut:
        shortcut_l.append([lr,val_auc,test_auc])       
    if not conv_bridge and not shortcut:
        none_of_them_l.append([lr,val_auc,test_auc])

both_of_them = np.array(both_of_them_l)
conv_bridge = np.array(conv_bridge_l)
shortcut = np.array(shortcut_l)
none_of_them = np.array(none_of_them_l)

this one for both of them! 0.01
       dataset     auc     acc  dice/F1  dice/F1_from_bin    spec    sens
0  DRIVE_train  0.9813  0.9568   0.8282            0.8274  0.9759  0.8245
1    DRIVE_val  0.9825  0.9544   0.8170            0.8061  0.9710  0.8347
2        DRIVE  0.9796  0.9559   0.8182            0.8219  0.9816  0.7797
3        STARE  0.9772  0.9579   0.7958            0.7825  0.9766  0.7950
4      CHASEDB  0.9669  0.9481   0.7374            0.7421  0.9737  0.7206
5      AV-WIDE  0.8482  0.9550   0.5841            0.5822  0.9893  0.4744


In [13]:
both_of_them_l = []
conv_bridge_l = []
shortcut_l = []
none_of_them_l = []
for i in range(len(cfgs)):
    conv_bridge = all_results[i][0]['conv_bridge']
    shortcut = all_results[i][0]['shortcut']
    lr = all_results[i][0]['lr']
    val_auc = all_results[i][1].iat[1,1]
    train_auc = all_results[i][1].iat[0,1]
    test_auc = all_results[i][1].iat[2,1]

    if not conv_bridge and not shortcut:
#         print('this one for none of them!', lr)
#         print(all_results[i][1])        
        none_of_them_l.append([lr,val_auc,test_auc])    
    
    if conv_bridge and not shortcut:
#         print('this one for ONLY conv_bridge!', lr)
#         print(all_results[i][1])
        conv_bridge_l.append([lr,val_auc,test_auc])
    
    if not conv_bridge and shortcut:
#         print('this one for ONLY shortcut!', lr)
#         print(all_results[i][1])
        shortcut_l.append([lr,val_auc,test_auc]) 
    


    if conv_bridge and shortcut:
        print('this one for both of them!', lr)
        print(all_results[i][1])
        both_of_them_l.append([lr,val_auc,test_auc])
    
both_of_them = np.array(both_of_them_l)
conv_bridge = np.array(conv_bridge_l)
shortcut = np.array(shortcut_l)
none_of_them = np.array(none_of_them_l)

this one for both of them! 0.01
       dataset     auc     acc  dice/F1  dice/F1_from_bin    spec    sens
0  DRIVE_train  0.9813  0.9568   0.8282            0.8274  0.9759  0.8245
1    DRIVE_val  0.9825  0.9544   0.8170            0.8061  0.9710  0.8347
2        DRIVE  0.9796  0.9559   0.8182            0.8219  0.9816  0.7797
3        STARE  0.9772  0.9579   0.7958            0.7825  0.9766  0.7950
4      CHASEDB  0.9669  0.9481   0.7374            0.7421  0.9737  0.7206
5      AV-WIDE  0.8482  0.9550   0.5841            0.5822  0.9893  0.4744


Best None Of Them

       dataset     auc     acc  dice/F1  dice/F1_from_bin    spec    sens
0  DRIVE_train  0.9799  0.9559   0.8243            0.8231  0.9755  0.8200
1    DRIVE_val  0.9825  0.9544   0.8172            0.8060  0.9707  0.8365
2        DRIVE  0.9791  0.9557   0.8196            0.8203  0.9797  0.7909
3        STARE  0.9667  0.9553   0.7816            0.7712  0.9759  0.7756
4      CHASEDB  0.9729  0.9494   0.7616            0.7479  0.9664  0.7984
5      AV-WIDE  0.8779  0.9557   0.6148            0.5862  0.9861  0.5302

Best ONLY CONV_BRIDGE

       dataset     auc     acc  dice/F1  dice/F1_from_bin    spec    sens
0  DRIVE_train  0.9805  0.9564   0.8262            0.8258  0.9758  0.8217
1    DRIVE_val  0.9826  0.9545   0.8178            0.8093  0.9708  0.8372
2        DRIVE  0.9793  0.9559   0.8203            0.8220  0.9800  0.7905
3        STARE  0.9774  0.9565   0.7922            0.7844  0.9742  0.8030
4      CHASEDB  0.9720  0.9491   0.7595            0.7517  0.9665  0.7943
5      AV-WIDE  0.8772  0.9546   0.6273            0.6000  0.9819  0.5728

Best ONLY SHORTCUT

       dataset     auc     acc  dice/F1  dice/F1_from_bin    spec    sens
0  DRIVE_train  0.9800  0.9558   0.8248            0.8245  0.9748  0.8240
1    DRIVE_val  0.9818  0.9542   0.8138            0.8050  0.9727  0.8208
2        DRIVE  0.9786  0.9555   0.8151            0.8207  0.9825  0.7705
3        STARE  0.9723  0.9562   0.7852            0.7745  0.9768  0.7766
4      CHASEDB  0.9655  0.9463   0.7201            0.7357  0.9759  0.6829
5      AV-WIDE  0.8530  0.9535   0.5909            0.5736  0.9857  0.5034

Best Both of Them

       dataset     auc     acc  dice/F1  dice/F1_from_bin    spec    sens
0  DRIVE_train  0.9813  0.9568   0.8282            0.8274  0.9759  0.8245
1    DRIVE_val  0.9825  0.9544   0.8170            0.8061  0.9710  0.8347
2        DRIVE  0.9796  0.9559   0.8182            0.8219  0.9816  0.7797
3        STARE  0.9772  0.9579   0.7958            0.7825  0.9766  0.7950
4      CHASEDB  0.9669  0.9481   0.7374            0.7421  0.9737  0.7206
5      AV-WIDE  0.8482  0.9550   0.5841            0.5822  0.9893  0.4744

In [14]:
none_of_them[none_of_them[:,0].argsort()]

array([[0.01  , 0.9825, 0.9787],
       [0.01  , 0.9825, 0.9791],
       [0.01  , 0.9816, 0.9784],
       [0.01  , 0.9823, 0.9789]])

In [17]:
conv_bridge[conv_bridge[:,0].argsort()]

array([[0.005 , 0.9823, 0.979 ],
       [0.005 , 0.9821, 0.9792],
       [0.005 , 0.9826, 0.9793],
       [0.005 , 0.9823, 0.9792],
       [0.01  , 0.9823, 0.9791],
       [0.01  , 0.9821, 0.979 ],
       [0.01  , 0.9823, 0.9791],
       [0.01  , 0.9821, 0.9788]])

In [18]:
shortcut[shortcut[:,0].argsort()]

array([[0.01  , 0.9818, 0.9786]])

In [15]:
both_of_them[both_of_them[:,0].argsort()]

array([[0.01  , 0.9825, 0.9796]])

In [ ]:
# conv_bridge[conv_bridge[:,0].argsort()]

In [ ]:
# shortcut[shortcut[:,0].argsort()]

In [ ]:
all_results[0][1]